examples/query/react/basic/src/tests/App.test.tsx

```diff
@@ -0,0 +1,74 @@
+import { render, screen, waitFor } from '@testing-library/react'
+import { Provider } from 'react-redux'
+import { rest } from 'msw'
+import { setupServer } from 'msw/node'
+import App from '../App'
+import { createStore } from '../store'
+
+const server = setupServer(
+  rest.get('https://pokeapi.co/api/v2/pokemon/bulbasaur', (req, res, ctx) => {
+    const mockApiResponse = {
+      species: {
+        name: 'bulbasaur',
+      },
+      sprites: {
+        front_shiny:
+          'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/1.png',
+      },
+    }
+    return res(ctx.json(mockApiResponse))
+  })
+)
+
+describe('App', () => {
+  beforeAll(() => server.listen())
+
+  afterEach(() => server.resetHandlers())
+
+  afterAll(() => server.close())
+
+  it('handles good response', async () => {
+    render(
+      <Provider store={createStore()}>
+        <App />
+      </Provider>
+    )
+
+    await screen.getByText('Loading...')
+
+    await waitFor(() => {
+      screen.getByText('bulbasaur')
+
+      const img = screen.getByAltText('bulbasaur') as HTMLImageElement
+      expect(img.src).toBe(
+        'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/1.png'
+      )
+    })
```

Comment:
```suggestion
    await screen.findByRole('heading', { name: /bulbasaur/i })
    const img = screen.getByRole('img', {  name: /bulbasaur/i}) as HTMLImageElement
    expect(img.src).toBe(
      'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/1.png'
    )
```

Where possible, *byRole queries are typically preferred, especially compared to *byAltText

https://testing-library.com/docs/queries/about/#priority

Since the image should exist as soon as you've got the 'bulbasaur' heading there, you don't need to have that part wrapped in the `waitFor`. This makes it clearer which bits you're waiting for, and which bits should be ready together